### PR TITLE
add io.github.beamerpresenter.beamerpresenter manifest

### PR DESCRIPTION
**Pull request by stiglers-eponym**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions.
- [X] All assets referenced in the manifest are redistributable by any party.
- [X] I am an upstream contributor to the project.
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### Extra information
The source points to a commit shortly after version 0.2.3 of beamerpresenter. The difference to v0.2.3 lies only in small adaptions required for flatpak/flathub.

This manifest defines building for Qt 5.15. It is also possible to build for Qt 6.3 with minor changes (adapt version numbers, enable Qt 6 in poppler and disable Qt 5 in poppler).

This manifest defines one possible configuration of BeamerPresenter. Other configurations with other PDF engines have not been tested with flatpak but might become available in the future.

### CODE REVIEW

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+app-id: io.github.beamerpresenter.beamerpresenter
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:1]: The source is fetched from https://github.com/stiglers-eponym/ which doesn't really match the app-id. And `https://github.com/beamerpresenter` doesn't exist.

tl;dr the app-id doesn't meet the requirements.

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+branch: master
+runtime: org.kde.Platform
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: beamerpresenter
+build-options:
+  strip: true
+  no-debuginfo: true
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:9]:
```suggestion
```

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+finish-args:
+  - "--socket=x11"
+  - "--socket=wayland"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:12]:
```suggestion
  - "--socket=fallback-x11"
  - "--socket=wayland"
```

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  - "--socket=pulseaudio"
+  - "--filesystem=xdg-documents"
+  - "--filesystem=xdg-download"
+  - "--filesystem=xdg-public-share"
+  - "--filesystem=xdg-desktop"
+  - "--filesystem=xdg-config/beamerpresenter"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:18]: why do you need this access to xdg-config?

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  - "--own-name=io.github.beamerpresenter.beamerpresenter"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:19]: the app-id which is a valid d-bus name is granted by default, you don't need this.

```suggestion
```

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+modules:
+  - name: poppler
+    buildsystem: cmake-ninja
+    config-opts:
+    - "-DCMAKE_INSTALL_PREFIX=/app"
+    - "-DCMAKE_INSTALL_LIBDIR=/app/lib"
+    - "-DCMAKE_INSTALL_INCLUDEDIR=/app/include"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:26]:
```suggestion
```

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    - "-DUSE_QTPDF=OFF"
+    - "-DUSE_EXTERNAL_RENDERER=OFF"
+    - "-DUSE_TRANSLATIONS=ON"
+    - "-DGIT_VERSION=OFF"
+    - "-DAPPID_NAME=io.github.beamerpresenter.beamerpresenter"
+    - "-DINSTALL_METAINFO=ON"
+    - "-DCMAKE_BUILD_TYPE=Release"
+    - "-DCMAKE_INSTALL_PREFIX=/app"
+    - "-DCMAKE_INSTALL_SYSCONFDIR=/app/etc"
+    - "-DCMAKE_INSTALL_DATAROOTDIR=/app/share"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:59]:
```suggestion
```

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    - "-DCMAKE_INSTALL_LIBDIR=/app/usr/lib"
+    - "-DCMAKE_INSTALL_INCLUDEDIR=/app/usr/include"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:61]: never do `/app/usr`. it's not fsb compliant and trigger some other bugs with appstream.

You shouldn't need to even have to set these anyway.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+build-options:
+  strip: true
+  no-debuginfo: true
+finish-args:
+  - "--socket=x11"
+  - "--socket=wayland"
+  - "--socket=pulseaudio"
+  - "--filesystem=xdg-documents"
+  - "--filesystem=xdg-download"
+  - "--filesystem=xdg-public-share"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:16]: why do you need `xdg-public-share`?

File: io.github.beamerpresenter.beamerpresenter.yml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    - "-DENABLE_UTILS=OFF"
+    - "-DENABLE_LIBCURL=OFF"
+    - "-DENABLE_QT5=ON"
+    - "-DENABLE_QT6=OFF"
+    - "-DENABLE_GLIB=OFF"
+    - "-DENABLE_GOBJECT_INTROSPECTION=OFF"
+    - "-DENABLE_GTK_DOC=OFF"
+    - "-DENABLE_CPP=OFF"
+    cleanup:
+    - "/bin"
```
Comment (hfiguiere) [inline io.github.beamerpresenter.beamerpresenter.yml:38]: you probably need to cleanup the other developement files.

File: io.github.beamerpresenter.beamerpresenter.yml (hunk 1 above)
Comment (stiglers-eponym) [reply io.github.beamerpresenter.beamerpresenter.yml:1]: Thank you for the very quick reply and for fixing some mistakes!
Since this will probably not be solved soon, I will close the PR for now. I'm sorry for wasting your time with this.

Comment (hfiguiere) [reply io.github.beamerpresenter.beamerpresenter.yml:1]: it should be `io.github.stiglers_eponym.BeamerPresenter`. That's what the documentation says. And can be fixed right here too.

Comment (nedrichards) [reply io.github.beamerpresenter.beamerpresenter.yml:1]: What kinds of problems are you thinking of?

Comment (stiglers-eponym) [reply io.github.beamerpresenter.beamerpresenter.yml:1]: It's not guaranteed that I will maintain this project forever. Defining the app-id seems quite permanent to me, so I wanted to avoid connecting it to a private account. But if this is common practice for projects that don't own a (sub)domain, I will follow the advice and use `io.github.stiglers_eponym.BeamerPresenter`.

**Comment by hfiguiere** [comment]
bot, build io.github.beamerpresenter.beamerpresenter

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/io.github.beamerpresenter.beamerpresenter.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
Validation of files failed
```
